src/Stratis.Bitcoin.Features.LightWallet/Blocks/LightWalletBlockStoreService.cs

```diff
@@ -0,0 +1,105 @@
+using Stratis.Bitcoin.Features.BlockStore;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Features.LightWallet.Blocks
+{
+    /// <inheritdoc/>
+    public sealed class LightWalletBlockStoreService : ILightWalletBlockStoreService
+    {
+        private IAsyncLoop asyncLoop;
+        private readonly IAsyncLoopFactory asyncLoopFactory;
+        private readonly IBlockRepository blockRepository;
+        private readonly IConsensusManager consensusManager;
+        private readonly ILogger logger;
+        private readonly INodeLifetime nodeLifetime;
+        private readonly IPrunedBlockRepository prunedBlockRepository;
+        private readonly StoreSettings storeSettings;
+
+        /// <inheritdoc/>
+        public ChainedHeader PrunedUpToHeaderTip { get; private set; }
+
+        public LightWalletBlockStoreService(
+            IAsyncLoopFactory asyncLoopFactory,
+            IBlockRepository blockRepository,
+            IPrunedBlockRepository prunedBlockRepository,
+            IConsensusManager consensusManager,
+            ILoggerFactory loggerFactory,
+            INodeLifetime nodeLifetime,
+            StoreSettings storeSettings)
+        {
+            this.asyncLoopFactory = asyncLoopFactory;
+            this.blockRepository = blockRepository;
+            this.prunedBlockRepository = prunedBlockRepository;
+            this.consensusManager = consensusManager;
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.nodeLifetime = nodeLifetime;
+            this.storeSettings = storeSettings;
+        }
+
+        /// <inheritdoc/>
+        public void Start()
+        {
+            this.PrunedUpToHeaderTip = this.consensusManager.Tip.GetAncestor(this.prunedBlockRepository.PrunedTip.Height);
+
+            this.asyncLoop = this.asyncLoopFactory.Run($"{this.GetType().Name}.{nameof(this.PruneBlocksAsync)}", async token =>
+            {
+                await PruneBlocksAsync();
+            },
+            this.nodeLifetime.ApplicationStopping,
+            repeatEvery: TimeSpans.TenSeconds);
+        }
+
+        /// <summary>
+        /// Delete blocks continuously from the back of the store.
+        /// </summary>
+        /// <returns>The awaited task.</returns>
+        private async Task PruneBlocksAsync()
+        {
+            if (this.blockRepository.TipHashAndHeight.Height < this.storeSettings.PruneBlockMargin)
+                return;
+
+            if (this.blockRepository.TipHashAndHeight.Height == (this.PrunedUpToHeaderTip?.Height ?? 0))
+                return;
+
+            if (this.blockRepository.TipHashAndHeight.Height < (this.PrunedUpToHeaderTip?.Height ?? 0 + this.storeSettings.PruneBlockMargin))
+                return;
+
+            var heightToPruneFrom = this.blockRepository.TipHashAndHeight.Height - this.storeSettings.PruneBlockMargin;
+            ChainedHeader startFrom = this.consensusManager.Tip.GetAncestor(heightToPruneFrom);
```

Comment:
this should be store tip not  consensusManager.Tip

Reply:
Fixed 👍